### PR TITLE
[GitLab] Added application, auth, and audit datastreams

### PR DESCRIPTION
**Pull request by tehbooom**
## Proposed commit message

Added [application](https://docs.gitlab.com/ee/administration/logs/#application_jsonlog), [auth](https://docs.gitlab.com/ee/administration/logs/#auth_jsonlog), and [audit](https://docs.gitlab.com/ee/administration/logs/#audit_jsonlog) datastreams

Added dashboards for application and auth datastreams

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- 

## Screenshots

gitlab-application-dashboard
gitlab-auth-dashboard

### CODE REVIEW

**Comment by efd6** [comment]
@tehbooom What version have you tested this on. This is failing with `Error: failed to setup system runner: can't install the package: there was an apply error: installation failed: can't install the package: could not zip-install package; API status code = 400; response body = {"statusCode":400,"error":"Bad Request","message":"Encountered 1 errors creating saved objects: [{\"type\":\"dashboard\",\"id\":\"gitlab-6b62c186-77b6-49f2-a820-ca377575bb96\",\"error\":{\"isBoom\":true,\"isServer\":false,\"data\":null,\"output\":{\"statusCode\":400,\"payload\":{\"message\":\"[attributes.controlGroupInput.showApplySelections]: definition for this key is missing: Bad Request\",\"statusCode\":400,\"error\":\"Bad Request\"},\"headers\":{}},\"type\":\"unknown\"}}]"}`. I see this locally as well.

**Comment by kcreddy** [comment]
I have got a similar error when I used `8.14.3` for creating dashboards, but the CI runs with minimum `kibana.version` i.e., `8.13.0`. I had to remove this key `attributes.controlGroupInput.showApplySelections` from the dashboards indicated by the error message.

**Comment by tehbooom** [comment]
@efd6 The elasticstack I tested on is `8.14.1`. Im trying what @kcreddy suggested to see if it that passes.

**Comment by kcreddy** [comment]
@tehbooom this time the error message is: 
`Error: can't install the package: could not zip-install package; API status code = 422; response body = {"statusCode":422,"error":"Unprocessable Entity","message":"Document \"gitlab-6b62c186-77b6-49f2-a820-ca377575bb96\" belongs to a more recent version of Kibana [10.2.0] when the last known version is [8.9.0]."}`

You will need to change from `"typeMigrationVersion": "10.2.0"` to `"typeMigrationVersion": "8.9.0"` inside dashboards. I totally forgot about that one earlier.

**Comment by tehbooom** [comment]
@kcreddy ahh I reverted back to 10.2.0 thinking it wasn't necessary.

File: packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      type: ip
+      ignore_missing: true
+      on_failure:
+          - remove:
+              field: gitlab.application.meta.remote_ip
+              ignore_missing: true
+          - append:
+              field: error.message
+              value: '{{{_ingest.on_failure_message}}}'
+  - dot_expander:
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml:156]: And for other datastreams

File: packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  - script:
+      tag: "removes_values_from_key"
+      lang: painless
+      source: >
+        if (ctx.gitlab?.application != null) {
+          def fieldsToRename = new ArrayList(ctx.gitlab.application.keySet());
+          for (fieldName in fieldsToRename) {
+            if (fieldName.endsWith('values')) {
+              def newField = fieldName.substring(0, fieldName.length() - 7);
+              if (ctx.gitlab.application.containsKey(fieldName)) {
```
[the remark this answers is not in the document]
Comment (tehbooom) [reply packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml:90]: Are referencing `if (ctx.gitlab.application.containsKey(fieldName)) {` ? If so I don't think it is necessary.

File: packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  - append:
+      tag: event_type_append
+      field: event.type
+      value:
+        - info
+  - append:
+      field: event.category
+      value:
+        - configuration
+      if: ' ctx.gitlab?.application?.message != null && !(ctx.gitlab?.application?.message.contains("Failed Login:") || ctx.gitlab?.application?.message.contains("Successful Login:"))'
```
[the remark this answers is not in the document]
Comment (tehbooom) [reply packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml]: Already added in processor at line `205`. Removed this duplicate

File: packages/gitlab/data_stream/auth/_dev/test/pipeline/test-auth.log-expected.json [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+                        "lon": 90.5
+                    }
+                },
+                "ip": "67.43.156.18"
+            },
+            "tags": [
+                "preserve_original_event"
+            ],
+            "url": {
+                "path": "/group/project.git/info/refs?service=git-upload-pack"
```
[the remark this answers is not in the document]
Comment (tehbooom) [reply packages/gitlab/data_stream/auth/_dev/test/pipeline/test-auth.log-expected.json]: I added a few processors to set a temporary `url.original` then used `uri_parts` processor to extract these values. You can see that here https://github.com/tehbooom/integrations/blob/db8a711481f4338132c137c799a24f3229b36dfc/packages/gitlab/data_stream/auth/elasticsearch/ingest_pipeline/default.yml#L99

File: packages/gitlab/data_stream/auth/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+      if: ctx.client?.ip != null
+  - append:
+      field: related.ip
+      value: '{{client.ip}}'
+      if: ctx.client?.ip != null
+      allow_duplicates: false
+  - append:
+      field: event.category
+      value:
+      - web
```
[the remark this answers is not in the document]
Comment (tehbooom) [reply packages/gitlab/data_stream/auth/elasticsearch/ingest_pipeline/default.yml]: The logs here are [logs based on requests](https://docs.gitlab.com/ee/administration/logs/#authlog) that trigger an "alert". The `api` category wouldn't fit here as its not always from an API. `intrusion_detection` wouldn't fit here either. I think `web` is the best category and then adding either `error` or `info` to `event.type`

File: packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+  - append:
+      field: event.category
+      value: configuration
+      if: ' ctx.gitlab?.application?.message != null && !(ctx.gitlab?.application?.message.contains("Failed Login:") || ctx.gitlab?.application?.message.contains("Successful Login:"))'
+  - remove:
+      field: event.original
+      tag: remove_original_event
+      if: ctx?.tags == null || !(ctx.tags.contains("preserve_original_event"))
+      ignore_failure: true
+      ignore_missing: true
```
Comment (kcreddy) [inline packages/gitlab/data_stream/application/elasticsearch/ingest_pipeline/default.yml:239]: Remove this processor as it is covered in fleet final pipeline.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼 Minor suggestion

**Pull request merged**